### PR TITLE
fix: bad tnx-id when rejoin cluster

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes https://emqx.atlassian.net/browse/EMQX-10462

### <samp>🤖 Generated by Copilot at 5bd60e6</samp>

This pull request enhances the cluster configuration synchronization mechanism by using transaction ids to track and apply changes across nodes. It also updates the versions of several applications to include bug fixes and improvements. The main files affected are `emqx_cluster_rpc.erl`, `emqx_conf_app.erl`, and `emqx_conf_schema.erl`.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_conf/src/emqx_conf_schema.erl [hunk 1]
```diff
@@ -684,18 +684,18 @@ fields("cluster_call") ->
             )},
         {"max_history",
             sc(
-                range(1, 500),
+                range(100, 10240),
```
Comment (HJianBo) [inline apps/emqx_conf/src/emqx_conf_schema.erl:687]: nit: There may be compatibility issues here? If the previous configuration used values from 1 to 99.

File: apps/emqx/src/emqx_app.erl [hunk 2]
```diff
@@ -65,16 +63,6 @@ set_config_loader(Module) when is_atom(Module) ->
 get_config_loader() ->
     application:get_env(emqx, config_loader, emqx).
```
Comment (zhongwencool) [inline apps/emqx/src/emqx_app.erl:65]: Remove cluster_rpc_init_tnx_id app env. 
Pass tnx_id from cluster_rpc:start_link function first args.

File: apps/emqx_conf/src/emqx_cluster_rpc.erl [hunk 3]
```diff
@@ -276,29 +284,46 @@ on_mria_stop(leave) ->
 on_mria_stop(_) ->
     ok.
 
+wait_for_cluster_rpc() ->
```
Comment (zhongwencool) [inline apps/emqx_conf/src/emqx_cluster_rpc.erl:287]: Write more useful logs when wait for cluster_rpc shard and tables.

File: apps/emqx_conf/src/emqx_cluster_rpc.erl [hunk 4]
```diff
@@ -276,29 +284,46 @@ on_mria_stop(leave) ->
...
 %% @private
-handle_continue(?CATCH_UP, State) ->
+handle_continue({?CATCH_UP, TnxId}, State = #{node := Node}) ->
     %% emqx app must be started before
     %% trying to catch up the rpc commit logs
     ok = wait_for_emqx_ready(),
+    ok = wait_for_cluster_rpc(),
+    %% The init transaction ID is set in emqx_conf_app after
+    %% it has fetched the latest config from one of the core nodes
+    ok = maybe_init_tnx_id(Node, TnxId),
```
Comment (zhongwencool) [inline apps/emqx_conf/src/emqx_cluster_rpc.erl:323]: Update the new tnx_id after `wait_for_cluster_rpc` successed.

File: apps/emqx_conf/src/emqx_cluster_rpc.erl [hunk 5]
```diff
@@ -388,7 +413,8 @@ read_next_mfa(Node) ->
                 }),
                 TnxId;
             [#cluster_rpc_commit{tnx_id = LastAppliedID}] ->
-                LastAppliedID + 1
+                OldestId = get_oldest_mfa_id(),
+                max(LastAppliedID + 1, OldestId)
```
Comment (zhongwencool) [inline apps/emqx_conf/src/emqx_cluster_rpc.erl:417]: If the commit's tnxid is clean up in cluster_mfa, we should follow up the oldest mfa_id.

**Review by HJianBo — APPROVED** [review]
LGTM

**Pull request merged**